### PR TITLE
Finished Stablecoin Example

### PR DESCRIPTION
**Pull request by kshitjee**
## Description
Finished original stablecoin [example](https://github.com/aptos-labs/aptos-core/pull/13173) with added features from AIP-73 and AIP-83. In particular, removed denylist SmartTable and implemented checks to verify the primary store is not frozen instead. Tested untransferable stores in the unit tests. 

## How Has This Been Tested?
Unit tests

## Key Areas to Review
Please review the unit tests and denylist related functions.

### CODE REVIEW

**Comment by lightmark** [comment]
why we have two stable coin examples PRs?

File: aptos-move/move-examples/fungible_asset/stablecoin/sources/usdk.move [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+        assert_not_denylisted(from);
+        assert_not_denylisted(to);
+
+        let expected_message = Approval {
+            owner: from,
+            nonce: account::get_sequence_number(from),
+            chain_id: chain_id::get(),
+            spender: signer::address_of(spender),
+            amount,
+        };
```
Comment (JohnChangUK) [inline aptos-move/move-examples/fungible_asset/stablecoin/sources/usdk.move:175]: We should have the `to` address specified in the challenge.

File: aptos-move/move-examples/fungible_asset/stablecoin/sources/usdk.move [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+    /// Mint new tokens to the specified account. This checks that the caller is a minter, the stablecoin is not paused,
+    /// and the account is not denylisted.
+    public entry fun mint(minter: &signer, to: address, amount: u64) acquires Management, Roles, State {
+        assert_not_paused();
+        assert_is_minter(minter);
+        assert_not_denylisted(to);
+
+        let management = borrow_global<Management>(usdk_address());
+        let tokens = fungible_asset::mint(&management.mint_ref, amount);
+        deposit(primary_fungible_store::ensure_primary_store_exists(to, metadata()), tokens, &management.transfer_ref);
```
Comment (JohnChangUK) [inline aptos-move/move-examples/fungible_asset/stablecoin/sources/usdk.move:213]: To generalize this contract, later, might be worth passing in the address of the metadata in the function, i.e. passing in `Object<Metadata>` and having relevant admin checks in the contract.

File: aptos-move/move-examples/fungible_asset/stablecoin/sources/usdk.move [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+    public fun usdk_address(): address {
+        object::create_object_address(&@stablecoin, ASSET_SYMBOL)
+    }
+
+    #[view]
+    public fun metadata(): Object<Metadata> {
+        object::address_to_object(usdk_address())
+    }
+
+    /// Called as part of deployment to initialize the stablecoin.
```
Comment (movekevin) [inline aptos-move/move-examples/fungible_asset/stablecoin/sources/usdk.move:96]: I'd suggest adding the steps here from your stablecoin doc

File: aptos-move/move-examples/fungible_asset/stablecoin/sources/usdk.move [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+        );
+
+        // Set ALL stores for the fungible asset to untransferable.
+        fungible_asset::set_untransferable(constructor_ref);
+
+        // All resources created will be kept in the asset metadata object.
+        let metadata_object_signer = &object::generate_signer(constructor_ref);
+        move_to(metadata_object_signer, Roles {
+            master_minter: @master_minter,
+            minters: vector::empty(),
```
Comment (movekevin) [inline aptos-move/move-examples/fungible_asset/stablecoin/sources/usdk.move:117]: nit: Use vector[] here instead

Comment (movekevin) [reply aptos-move/move-examples/fungible_asset/stablecoin/sources/usdk.move:117]: Also why changed this from vector[@minter]. I still see the minter address in Move.toml

**Review by movekevin — APPROVED** [review]

File: aptos-move/move-examples/fungible_asset/stablecoin/sources/usdk.move (hunk 4 above)
Comment (kshitjee) [reply aptos-move/move-examples/fungible_asset/stablecoin/sources/usdk.move:117]: Wanted to test with the add_minter function in unit tests

**Review by runtian-zhou — COMMENTED** [review]
Left a few suggestions for comments. Overall LGTM

**Review by runtian-zhou — APPROVED** [review]

File: aptos-move/move-examples/fungible_asset/stablecoin/sources/tests/usdk_tests.move [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        assert!(primary_fungible_store::is_frozen(receiver_address, asset), 0);
+
+        let constructor_ref = object::create_object(receiver_address);
+        fungible_asset::create_store(&constructor_ref, asset);
+        let store = object::object_from_constructor_ref<FungibleStore>(&constructor_ref);
+
+        object::transfer(receiver, store, @0xdeadbeef);
+    }
+
+
```
Comment (JohnChangUK) [inline aptos-move/move-examples/fungible_asset/stablecoin/sources/tests/usdk_tests.move:66]: Nit: remove these empty lines

**Review by JohnChangUK — APPROVED** [review]

**Pull request merged**